x-pack/plugins/transform/public/app/hooks/use_pivot_data.ts

```diff
@@ -79,12 +79,16 @@ export function getCombinedProperties(
   populatedProperties: PreviewMappingsProperties,
   docs: Array<Record<string, unknown>>
 ): PreviewMappingsProperties {
-  // Take a sample from docs and resolve missing mappings
-  const sampleDoc = sample(docs) ?? {};
-  const missingMappings = difference(Object.keys(sampleDoc), Object.keys(populatedProperties));
+  // Identify missing mappings
+  const missingMappings = difference(
```

Comment:
The grid isn't inferring the type of the value correctly - it thinks it's a string, so left aligns the cell contents, and gives the alphabetical sort options (note sort isn't working on nested fields but I will raise a separate issue for that).

![image](https://user-images.githubusercontent.com/7405507/151021046-3ce7d1c8-fcb9-425d-b948-f5a08ce333f1.png)

Reply:
It makes sense to address this as part of https://github.com/elastic/kibana/issues/123796 as even if the type for the `terms` agg columns is correctly set to numeric it still won't be sorted correctly as the column name will contain a dot.